File: swift/hamming/Hamming.swift
import Foundation
class Hamming {
    class func compute(firstString: String, against: String) -> Int? {
        
        let firstCharacters: [Character] = firstString.characters.map { $0 }
        let againstCharacters: [Character] = against.characters.map { $0 }
        
        guard firstCharacters.count == againstCharacters.count else { return nil }
        
        var count = 0
        for (index, character) in firstCharacters.enumerate() {
            if character != againstCharacters[index] {
                count+=1
            }
        }
        return count
    }
}

File: ruby/hamming/hamming.rb

class Hamming
  VERSION = 2
  def self.compute(string_one, string_two)
    raise ArgumentError if string_one.length != string_two.length
    index = 0
    counter = 0
  while (first = string_one[index]) && (second = string_two[index])
    counter += 1 if first != second
    index +=1
  end
  return counter
  end
end
